connect-inject/metrics_configuration.go

```diff
@@ -0,0 +1,194 @@
+package connectinject
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type MetricsConfiguration struct {
+	DefaultEnableMetrics        bool
+	DefaultEnableMetricsMerging bool
+	DefaultMergedMetricsPort    string
+	DefaultPrometheusScrapePort string
+	DefaultPrometheusScrapePath string
+}
+
+const (
+	defaultServiceMetricsPath = "/metrics"
+)
+
+// mergedMetricsServerConfiguration is called when running a merged metrics server and used to return ports necessary to
+// configure the merged metrics server.
+func (mc MetricsConfiguration) mergedMetricsServerConfiguration(pod corev1.Pod) (string, string, string, error) {
```

Comment:
oh one thing I forgot to say is that at the very least, each return value needs to be documented here and it might make sense to actually return a struct here, e.g. `MetricsPorts` with fields `MergedPort`, `ServicePort`, `MetricsPath`.

Reply:
good idea, I changed this to use the struct. It could be unexported since it's only used within the package.